apache/zeppelin
# ZEPPELIN-3072: Zeppelin UI becomes slow/unresponsive if there are too many notebooks

### What is this PR for?
Zeppelin UI becomes slow/unresponsive if there are too many notebooks

Have attached a notebook directory in [JIRA](https://issues.apache.org/jira/secure/attachment/12898650/notebook.zip) with 500+ notebooks, now with these notebooks, every time user goes to homepage Zeppelin UI becomes unresponsive for few seconds.

### What type of PR is it?
[Bug Fix | Improvement]

### Todos
* [x] - Fix search box
* [x] - Order of notebook

### What is the Jira issue?
* [ZEPPELIN-3072](https://issues.apache.org/jira/browse/ZEPPELIN-3072)

### How should this be tested?
 Create 500+ notebook, or import it from [JIRA](https://issues.apache.org/jira/secure/attachment/12898650/notebook.zip), now observe UI it becomes slow/laggy while homepage is rendering. 

### Screenshots (if appropriate)

Before:
![before](https://user-images.githubusercontent.com/674497/33070354-c36acdfa-cedd-11e7-81f9-ff0b526622f3.gif)

After:
![after](https://user-images.githubusercontent.com/674497/33070353-c3317988-cedd-11e7-9431-fcf596928c3b.gif)



### Questions:
* Does the licenses files need update? N/A
* Is there breaking changes for older versions? N/A
* Does this needs documentation? N/A
